Source/ablastr/fields/PoissonSolver.H

```diff
@@ -153,6 +160,23 @@ computePhi (amrex::Vector<amrex::MultiFab*> const & rho,
                 {{ beta[0], beta[1], beta[2] }};
 #endif
 
+#if (defined(WARPX_USE_PSATD)  && defined(WARPX_DIM_3D))
+        // Use the Integrated Green Function solver (fft-based) on the coarsest level if it was selected
+        if(WarpX::poisson_solver_id==PoissonSolverAlgo::IntegratedGreenFunction && lev==0){
```

Comment:
Instead of relying on global variables of `WarpX::`, please pass parameters explicit in ABLASTR.

Add a new parameter or enum parameter to `computePhi` that selects the algorithm.

When you call `computePhi` inside WarpX you can use `WarpX::poisson_solver_id==PoissonSolverAlgo::IntegratedGreenFunction`, but not inside ABLASTR itself, because ImpactX will not have access to `WarpX::` globals.